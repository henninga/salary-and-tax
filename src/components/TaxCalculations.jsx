import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import PropTypes from 'prop-types';
import React, { Fragment } from 'react';
import { defineMessages, injectIntl, intlShape } from 'react-intl';
import { connect } from 'react-redux';
import { formatCurrency } from '../calculations';
import { Section, TaxRow } from './Ui';


const messages = defineMessages({
  header: {id: 'tax.calculations.header', defaultMessage: 'Tax calculations'},
  income: {id: 'tax.calculations.income', defaultMessage: 'Income'},
  minimumDeduction: {id: 'tax.calculations.minimum.deduction', defaultMessage: 'Minimum deduction'},
  commonIncome: {id: 'tax.calculations.common.income', defaultMessage: 'Common income'},
  incomeTax: {id: 'tax.calculations.income.tax', defaultMessage: 'Income tax'},
  socialSecurityDeduction: {id: 'tax.calculations.social.security.deduction', defaultMessage: 'Social security deduction'},
  step1: {id: 'tax.calculations.step.1', defaultMessage: 'Tax step 1'},
  step2: {id: 'tax.calculations.step.2', defaultMessage: 'Tax step 2'},
  step3: {id: 'tax.calculations.step.3', defaultMessage: 'Tax step 3'},
  step4: {id: 'tax.calculations.step.4', defaultMessage: 'Tax step 4'},
  totalTax: {id: 'tax.calculations.total.tax', defaultMessage: 'Total tax'},
});

export const TaxCalculationsComponent = ({
  income, minimumDeduction, commonIncome, incomeTax, socialSecurityDeduction, step1, step2, step3, step4, totalTax, intl: {formatMessage},
}) => {
  return (
    <Fragment>
      <Section header={formatMessage(messages.header)}>
        <Table>
          <TableBody>
            <TaxRow description={formatMessage(messages.income)} amount={formatCurrency(income)} />
            <TaxRow description={formatMessage(messages.minimumDeduction)} amount={formatCurrency(minimumDeduction)} minus />
            <TaxRow description={formatMessage(messages.commonIncome)} amount={formatCurrency(commonIncome)} />
          </TableBody>
        </Table>

        <Table>
          <TableBody>
            <TaxRow description={formatMessage(messages.commonIncome)} amount={formatCurrency(commonIncome)} />
            <TaxRow description={formatMessage(messages.incomeTax)} amount={formatCurrency(incomeTax)} minus />
            <TaxRow description={formatMessage(messages.socialSecurityDeduction)} amount={formatCurrency(socialSecurityDeduction)} minus />
            <TaxRow description={formatMessage(messages.step1)} amount={formatCurrency(step1)} minus />
            <TaxRow description={formatMessage(messages.step2)} amount={formatCurrency(step2)} minus />
            <TaxRow description={formatMessage(messages.step3)} amount={formatCurrency(step3)} minus />
            <TaxRow description={formatMessage(messages.step4)} amount={formatCurrency(step4)} minus />
            <TaxRow description={formatMessage(messages.totalTax)} amount={formatCurrency(totalTax)} />
          </TableBody>
        </Table>
      </Section>
    </Fragment>
);
};

TaxCalculationsComponent.propTypes = {
  income: PropTypes.number.isRequired,
  minimumDeduction: PropTypes.number.isRequired,
  commonIncome: PropTypes.number.isRequired,
  incomeTax: PropTypes.number.isRequired,
  socialSecurityDeduction: PropTypes.number.isRequired,
  step1: PropTypes.number.isRequired,
  step2: PropTypes.number.isRequired,
  step3: PropTypes.number.isRequired,
  step4: PropTypes.number.isRequired,
  totalTax: PropTypes.number.isRequired,
  intl: intlShape.isRequired,
};

const mapStateToProps = ({calculations: {tax}}) => ({...tax});

const mapDispatchToProps = () => ({});

export const TaxCalculations = connect(mapStateToProps, mapDispatchToProps)(injectIntl(TaxCalculationsComponent));
export default TaxCalculations;
