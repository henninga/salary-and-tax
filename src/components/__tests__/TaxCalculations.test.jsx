import mockComponent from 'identity-obj-proxy'
import React from 'react';
import renderer from 'react-test-renderer';
import { TaxCalculationsComponent } from '../TaxCalculations';


jest.mock('../Ui', () => mockComponent);
jest.mock('@material-ui/core/Table', () => "Table");
jest.mock('@material-ui/core/TableBody', () => "TableBody");


describe('TaxCalculations', () => {
  it('should render', () => {
    const tree = renderer.create(<TaxCalculationsComponent
      income={12345678}
      minimumDeduction={12345678}
      commonIncome={12345678}
      incomeTax={12345678}
      socialSecurityDeduction={12345678}
      step1={12345678}
      step2={12345678}
      step3={12345678}
      step4={12345678}
      totalTax={12345678}
      afterTax={12345678}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
