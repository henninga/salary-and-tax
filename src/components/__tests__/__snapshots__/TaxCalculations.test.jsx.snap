// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaxCalculations should render 1`] = `
<Section
  header="Tax calculations"
>
  <Table>
    <TableBody>
      <TaxRow
        description="Income"
        sum="$12,345,678.00"
      />
      <TaxRow
        amount="$12,345,678.00"
        description="Minimum deduction"
        minus={true}
      />
      <TaxRow
        description="Common income"
        sum="$12,345,678.00"
      />
    </TableBody>
  </Table>
  <Table>
    <TableBody>
      <TaxRow
        description="Income"
        sum="$12,345,678.00"
      />
      <TaxRow
        amount="$12,345,678.00"
        description="Income tax"
        minus={true}
      />
      <TaxRow
        amount="$12,345,678.00"
        description="Social security deduction"
        minus={true}
      />
      <TaxRow
        amount="$12,345,678.00"
        description="Tax step 1"
        minus={true}
      />
      <TaxRow
        amount="$12,345,678.00"
        description="Tax step 2"
        minus={true}
      />
      <TaxRow
        amount="$12,345,678.00"
        description="Tax step 3"
        minus={true}
      />
      <TaxRow
        amount="$12,345,678.00"
        description="Tax step 4"
        minus={true}
      />
      <TaxRow
        amount="$12,345,678.00"
        description="After tax"
        sum="$12,345,678.00"
      />
    </TableBody>
  </Table>
</Section>
`;
